### ISSUE TITLE
Convert Serum project into a Mix project

### ISSUE THREAD

**Issue by Dalgona** [Dalgona/Serum#26]
Serum uses its own directory structure and file layout for each Serum project, but nowadays I think if a Serum project is also compatible with Mix, it would enable us to do much more useful things as we author our websites using Serum.

## Rationale

The main reason for this is the lack of package management. If we could get some packages from Hex for each Serum project locally, we could add some bells and whistles such as "Serum plugins™" and "Serum themes™." However, if we want to install a package locally, we must do it in a Mix project, and [there is no global install in Elixir](https://stackoverflow.com/a/33568506) (note the difference from `mix archive.install`).

## So, what will change?

- Obviously this new version of Serum won't be compatible with every existing Serum projects, but it won't be that hard to migrate your projects, I guess. (I'm not sure there will be a migration tool at that moment)
- There will be no `serum` escript any more. The preferred installation method will be `mix archive.install`, and you will be able to manage Serum projects using Mix tasks (just like `mix phx.new` in Phoenix Framework).
- And much more, maybe.

**Comment by Dalgona** [comment]
Current Serum project layout:

```
/path/to/project
├─ assets/
├─ includes/
├─ media/
├─ pages/
├─ posts/
├─ templates/
│  ├─ base.html.eex
│  ├─ list.html.eex
│  ├─ page.html.eex
│  └─ post.html.eex
├─ .gitignore
└─ serum.json
```

Typical Mix project layout:

```
/path/to/project
├─ config/
│  └─ config.exs
├─ lib/
├─ priv/ (optional)
├─ test/
│  └─ test_helper.exs
├─ .formatter.exs
├─ .gitignore
└─ mix.exs
```

## Useful materials

- Source code of [`Mix.Tasks.New.run/1`](https://github.com/elixir-lang/elixir/blob/v1.8.1/lib/mix/lib/mix/tasks/new.ex#L59)
- [Mix.Generator](https://hexdocs.pm/mix/Mix.Generator.html) module
- [How to get project's root dir path](https://groups.google.com/forum/#!topic/elixir-lang-talk/Ls0eJDdMMW8)

**Comment by Dalgona** [comment]
After some experiment, I figured out that the minimum requirement for Mix to work in a directory is just a valid `mix.exs` file. That means if we just add `mix.exs` in the existing Serum project directory and we can take advantage of Mix build tool.

So the new Serum project would look like:

```
/path/to/project
├─ assets/
├─ includes/
├─ media/
├─ pages/
├─ posts/
├─ templates/
│  ├─ base.html.eex
│  ├─ list.html.eex
│  ├─ page.html.eex
│  └─ post.html.eex
├─ lib/           (optional)
├─ priv/          (optional)
├─ .formatter.exs (optional)
├─ .gitignore
├─ mix.exs
└─ serum.json
```

We can then refer to these files using `File.cwd!/0` inside Mix tasks. ([source](https://groups.google.com/d/msg/elixir-lang-talk/Ls0eJDdMMW8/VLWWAKWPAQAJ))

With these information, I will start working on implementing `Mix.Tasks.Serum.New` Mix task module.

**Comment by Dalgona** [comment]
The conversion is now done and I published `serum-0.8.1` and `serum_new-0.8.1` to Hex. There is some more works to be done though (mostly docs). Refer to the latest version of README.md for installation guide.
